controller/internal/enforcer/nfqdatapath/afinetrawsocket/afinetrawsocket.go

```diff
@@ -41,7 +41,7 @@ const (
 
 // SocketWriter interface exposes an interface to write and close sockets
 type SocketWriter interface {
-	WriteSocket(buf []byte, version packet.IPver) error
+	WriteSocket(buf []byte, version packet.IPver, data interface{}) error
```

Comment:
I would prefer no generic interface definitions in the methods. Any reason we actually need an interface and we cannot define a datatype.

Reply:
The metadata is stored on the packet.Packet structure, and it's an interface{} there following similar conventions for other members on that struct (due to circular reference?). But I can clean up this interface and make it take a concrete type.